src/main/java/bio/overture/ego/utils/Redirects.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2019. The Ontario Institute for Cancer Research. All rights reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package bio.overture.ego.utils;
+
+import static java.lang.String.format;
+
+import bio.overture.ego.model.entity.Application;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.Objects;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.springframework.security.oauth2.common.exceptions.UnauthorizedClientException;
+
+@Slf4j
+public class Redirects {
+
+  /**
+   * Returns redirect uri based Declares a runtime exception to be explicit
+   *
+   * @return Returns URI as a String that Ego will redirect to
+   */
+  public static String getRedirectUri(@NonNull Application app, String redirect) {
+    val redirects = Arrays.stream(app.getRedirectUri().split(","));
+
+    // Short return if no redirect URI is provided
+    if (redirect == null || redirect.isEmpty()) {
+      return redirects
+          .findFirst()
+          .orElseThrow(() -> new UnauthorizedClientException("Cannot find valid redirect URI"));
+    }
+
+    val msg = format("Unauthorized redirect URI: %s", redirect);
+    URI redirectUri;
+    try {
+      redirectUri = new URI(redirect);
+    } catch (URISyntaxException e) {
+      log.error(msg);
+      throw new UnauthorizedClientException(msg);
```

Comment:
are those unauthorized or bad request ?

Reply:
I'd argue unauthorized as by the time it gets here, I've understood the request.